[HER-171] Update the design of the characters section

andreea-dumitrascu opened the pull request:
## Description
We should add on the home page the design for two characters that are displayed in parallel like in the photo: 
![image](https://user-images.githubusercontent.com/72592844/137962583-0a5902dd-b51f-4524-8c1b-7514d6cd3c0b.png)
[Desktop](https://xd.adobe.com/view/f1281b54-d458-42c1-b28b-7dcca2c23b31-a47b/specs/?fbclid=IwAR3x9zSAwHJJIrQaGxIQvAKjAwys50dFidZfz7ZnOpH4-rzQX-W-kQO4LVo), [Tablet](https://xd.adobe.com/view/c08e8b21-5cdf-469b-9e66-2c500a5ec043-95dd/?fbclid=IwAR1x4Wn89kjsO0ZivtrxqBztZvh67ubeCMJGVc8eMmivnjwBRwteIr7zP20), [Phone](https://xd.adobe.com/view/c5fc8477-17c8-4014-872e-6e6575603d84-90af/?fbclid=IwAR2aCw3az4LwYWh4oYOW08rfN_9r8F8xh8u98QghtsQUYmHNQoKX8QUCAt4)

## Related
[HER-171](https://nerds-interns.atlassian.net/jira/software/projects/HER/boards/1?selectedIssue=HER-171)

andreea-dumitrascu committed:
[HER-171] Delete Slider component

andreea-dumitrascu committed:
[HER-171] Create characters content

andreea-dumitrascu committed:
[HER-171] Create the content component

andreea-dumitrascu committed:
[HER-171] Create the style for the title of the content

andreea-dumitrascu committed:
[HER-171] Create the style for the description
of the content

andreea-dumitrascu committed:
[HER-171] Create character component

andreea-dumitrascu committed:
[HER-171] Create the character`s wrapper style

andreea-dumitrascu committed:
[HER-171] Create the style for the image

andreea-dumitrascu committed:
[HER-171] Create the props for the image

andreea-dumitrascu committed:
[HER-171] Create the style for the desktop mode

andreea-dumitrascu committed:
[HER-171] Create the style for the mobile mode

andreea-dumitrascu committed:
[HER-171] Create the style for the box from character

andreea-dumitrascu committed:
[HER-171] Create the style for the left component

andreea-dumitrascu committed:
[HER-171] Create the style for the right component

andreea-dumitrascu committed:
[HER-171] Create the style for the grid from character

andreea-dumitrascu committed:
[HER-171] Create the style for the subtitle of home page

andreea-dumitrascu committed:
[HER-171] Create the style of the picture`s border

andreea-dumitrascu committed:
[HER-171] Create the character section

andreea-dumitrascu committed:
[HER-171] Add the character section on home page

miutamihai commented on src/home/character-section/hooks/use-class-name.js:
```diff
@@ -0,0 +1,7 @@
+export const useClassName = position => {
+  const style = useStyles()
+
+  return ({ className: position === 'left' ? style.left : style.right })
```
Instead of returning an object containing a field called `className`, why don't we return just the className here?

miutamihai replied:
Also, it should be `future` instead of `feature` characters.

ConstantinescuAndrei replied:
Done

miutamihai replied:
No, it's not.

miutamihai commented on src/home/character-section/hooks/use-character-title-style.js:
```diff
@@ -0,0 +1,20 @@
+const style = theme => ({
+  root: {
+    ...common,
+    [theme.breakpoints.up('xs')]: { ...desktop },
```
Using the spread operator for a single object field is overkill. At this point, we're adding complexity to our code instead of reducing it. Why not pass in the __value__ we need directly?

ConstantinescuAndrei replied:
Done

miutamihai commented on src/home/character-section/hooks/use-character-title-style.js:
```diff
@@ -0,0 +1,20 @@
+  root: {
+    ...common,
+    [theme.breakpoints.up('xs')]: { ...desktop },
+    [theme.breakpoints.down('xs')]: { ...phone },
```
Same here.

ConstantinescuAndrei replied:
Done

miutamihai commented on src/home/character-section/hooks/use-descripiton-style.js:
```diff
@@ -0,0 +1,14 @@
+
+const style = theme => ({
+  root: {
+    [theme.breakpoints.up('xs')]: { ...desktop },
```
Same here.

miutamihai replied:
Also there's a typo in the file name.

ConstantinescuAndrei replied:
Done

miutamihai commented on src/home/character-section/hooks/use-left-style.js:
```diff
@@ -0,0 +1,28 @@
+import { makeStyles } from '@material-ui/core'
+
+const desktop = ({
```
![image](https://user-images.githubusercontent.com/23569938/138052809-2cb0a983-c5ff-4b60-8f38-484f5f08a09c.png)
These two files are identical with the exception of the `marginLeft` prop. Could we merge them somehow?

ConstantinescuAndrei replied:
Done

miutamihai commented on src/home/character-section/styles.js:
```diff
@@ -0,0 +1,14 @@
+
+const right = {
+  float: 'right',
+  borderRight: `5px solid ${theme.palette.secondary.main}`,
```
We've talked previously about importing the `theme` directly. Let's fix this.

ConstantinescuAndrei replied:
Done

miutamihai replied:
![image](https://user-images.githubusercontent.com/23569938/138274386-e5c51b1a-5208-4259-9e61-e08dd3be25fc.png)
Let's rename this `create` function to something like `makeForVariant` or `styleForVariant`.

miutamihai reviewed (changes requested):
![image](https://user-images.githubusercontent.com/23569938/138053576-3fbf1e2b-a151-4702-8eb2-0537f9240500.png)
Let's replace these placeholders with simple fixed sized divs having the required css.

ConstantinescuAndrei committed:
[HER-171] Follow-up Extract in components

- extract Grid in Wrapper component
- extract Title
- extract Characters

ConstantinescuAndrei committed:
[HER-171] Follow-up Move items file

ConstantinescuAndrei committed:
[HER-171] Follow-up Move character component

Move Character component under characters/character directory
Extract Wrapper, Image and text components

ConstantinescuAndrei committed:
[HER-171] Follow-up Move hook

ConstantinescuAndrei committed:
[HER-171] Follow-up Create Image component

ConstantinescuAndrei committed:
[HER-171] Create styles

Create styles for position left and right
Create hook for returning the correct style

ConstantinescuAndrei committed:
[HER-171] Follow-up Create Text component

ConstantinescuAndrei committed:
[HER-171] Follow-up Create styles for text component

ConstantinescuAndrei committed:
[HER-171] Follow-up Create custom hooks

Create custom hooks for choosing the right style depending of the position

ConstantinescuAndrei committed:
[HER-171] Follow-up move component under content

ConstantinescuAndrei committed:
[HER-171] Follow-up Create styles for content

ConstantinescuAndrei committed:
[HER-171] Follow-up Create hooks

Create hooks for choosing the correct style for title and description
The style will be chosen depending on the screen size

ConstantinescuAndrei committed:
[HER-171] Follow-up Remove unused files

ConstantinescuAndrei committed:
[HER-171] Follow-up Add new color to palette

ConstantinescuAndrei committed:
[HER-171] Follow-up Create style for image

ConstantinescuAndrei committed:
[HER-171] Follow-up Remove unused styles

ConstantinescuAndrei committed:
[HER-171] Follow-up Rename file

Urzica-Platon-Casian reviewed (approved):
Nois
